bokeh/application/handlers/directory.py

```diff
@@ -32,6 +36,10 @@ def url_path(self):
     def modify_document(self, doc):
         if self.failed:
             return
+        # Note: we do NOT copy self._theme, which assumes the Theme
+        # class is immutable (has no setters)
```

Comment:
It would be nice to enforce this assumption but that can happen in more `Theme` development later.